src/Resolver/Factory.php

```diff
@@ -62,4 +74,27 @@ protected function addPortToServerIfMissing($nameserver)
 
         return $nameserver;
     }
+
+    protected function determineSystemDefaultServer()
+    {
+        $nameservers = array();
+
+        if (stripos(PHP_OS, 'win') === false) {
+            $lines = file(self::UNIX_RESOLVE_CONF);
+            foreach ($lines as $line) {
+                $nameserverPosition = stripos($line, 'nameserver');
+                if ($nameserverPosition !== false) {
+                    $nameserverLine = trim(substr($line, $nameserverPosition + 11));
+                    list ($nameservers[]) = explode(' ', $nameserverLine);
+                }
+            }
+        }
+
+        if (count($nameservers) > 0) {
+            shuffle($nameservers);
```

Comment:
I think it should use the first one, not a random one. I think the first one is considered the primary DNS and order should be respected in `/etc/resolv.conf`.